components/kubeflow-pipeline/pipeline_components/flwr-server/flwr_server/__main__.py

```diff
@@ -137,7 +148,8 @@
         confusion_csv = io.BytesIO()
         df_cm.to_csv(confusion_csv, header=False, index=False)
 
-        metadata["outputs"].append(
+        outputs: List[Any] = cast(List[Any], metadata["outputs"])
+        outputs.append(
```

Comment:
I suppose the `outputs` intermediary variable is used to do the typing right. If so, I would suggest to fix this on the `metadata`, e.g:

```python
import dataclasses

@dataclasses.dataclass
class Metadata:
    version: int
    # We could go a step forward and remove the Any but I think it is ok for the time being without it.
    outputs: List[Dict[str, Any]]

# ...

metadata = Metadata(
    version=1,
    outputs=[
        {
            "type": "table",
            "format": "csv",
            "header": [
                "run",
                "client_id",
                "accuracy",
                "precision",
                "recall",
                "num_examples",
            ],
            "storage": "inline",
            "source": metrics_csv.getvalue().decode("utf-8"),
        },
    ],
)

# ...

metadata.outputs.append(
    {
        "type": "confusion_matrix",
        "format": "csv",
        "schema": [
            {"name": "target", "type": "CATEGORY"},
            {"name": "predicted", "type": "CATEGORY"},
            {"name": "count", "type": "NUMBER"},
        ],
        "labels": df_cm.columns.tolist(),
        "storage": "inline",
        "source": confusion_csv.getvalue().decode("utf-8"),
    }
)

# ...

Path(args.metadata_output_path).write_text(
    json.dumps(dataclasses.asdict(metadata))
)

```

Reply:
I wanted to avoid introducing types here and basically treat `metadata` as if it's already (untyped) JSON. By having the cast, this is saying "I know that this is a list, let's treat it like that".

Reply:
I see. OTOH we might endup implementing that typing anyways because it needs to have a specific set of fields. That's why I think it is not too terrible to introduce this. Although I think this discussion can be continued in a different PR because this is a broader topic regarding the typing and to which extend apply it. Therefore I'll mark this comment as resolved.